usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/port/GtfsDataRepository.java

```diff
@@ -71,4 +72,8 @@
 
     FareRule getFareRule(final String fareId, final String routeId, final String originId, final String destinationId,
                          final String containsId);
+
+    ShapePoint addShapePoint(final ShapePoint newShapePoint) throws IllegalArgumentException;
+
+    TreeMap<Integer, ShapePoint> getShapeById(final String shapeId);
```

Comment:
I would add Javadocs here saying that the returned map is ordered by shape point sequence.

Reply:
Not against it, but wouldn't that duplicate the javadoc written in `InMemoryGtfsDataRepository`?
Also, to ensure consistency if we add javadoc here, we should add javadoc for all methods of this interface (maybe to all interface?).

Reply:
Yes, but for an interface if there are implementation assumptions like an ordered map that should be documented in the interface too. Otherwise later someone may come along and implement the interface again, but not realized that there is an assumption that the method returns ordered results. `TreeMap` implies this, but someone could generalize this to `Map` at some point as well, also not realizing that order matters.

In other projects I've ended up duplicating Javadocs across interfaces and implementations in some cases. It's most important when there are assumptions that may not be obvious.